agent/engine/ordering_integ_test.go

```diff
@@ -339,6 +339,72 @@ func TestDependencyHealthyTimeout(t *testing.T) {
 	waitFinished(t, finished, orderingTimeout)
 }
 
+
+func TestGranularStopTimeout(t *testing.T) {
+	taskEngine, done, _ := setupWithDefaultConfig(t)
+	defer done()
+
+	stateChangeEvents := taskEngine.StateChangeEvents()
+
+	taskArn := "testDependencyHealthyTimeout"
+	testTask := createTestTask(taskArn)
+
+	dependency := createTestContainerWithImageAndName(baseImageForOS, "dependency")
```

Comment:
I'd do this test the other way around -- 1 parent, 2 dependencies. You can get rid of the sleep (and do exit 0 instead) since this will guarantee that the other containers have started.

Reply:
Makes sense.